lib/modules/manager/kustomize/artifacts.ts

```diff
@@ -0,0 +1,261 @@
+import is from '@sindresorhus/is';
+import semver from 'semver';
+import { quote } from 'shlex';
+import upath from 'upath';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type {
+  ExecOptions,
+  ExtraEnv,
+  ToolConstraint,
+} from '../../../util/exec/types';
+import {
+  deleteLocalFile,
+  getSiblingFileName,
+  localPathExists,
+  privateCacheDir,
+  readLocalFile,
+} from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import { DockerDatasource } from '../../datasource/docker';
+import { HelmDatasource } from '../../datasource/helm';
+import type {
+  UpdateArtifact,
+  UpdateArtifactsConfig,
+  UpdateArtifactsResult,
+} from '../types';
+import { parseKustomize } from './extract';
+
+function generateHelmEnvs(config: UpdateArtifactsConfig): ExtraEnv {
+  const cacheDir = privateCacheDir();
+  const enableExperimentalOci =
+    config.constraints?.helm &&
+    semver.satisfies(config.constraints.helm, '<3.8.0');
+
+  return {
+    HELM_EXPERIMENTAL_OCI: enableExperimentalOci ? '1' : '0',
```

Comment:
don't add this env at all on newer helm. needs a test.

you probably should move this function to common or utils file in this folder for easier testing